WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1392,7 +1380,8 @@ extension ReaderStreamViewController: WPTableViewHandlerDelegate {
     // MARK: - Fetched Results Related
 
     func managedObjectContext() -> NSManagedObjectContext {
-        return displayContext
+        assert(Thread.isMainThread, "WPTableViewHandler should use Core Data on the main thread")
```

Comment:
Quick safety check – I don't _think_ `WPTableViewHandler` is going to act up, but if it does I'd love to catch it in action while working on this project.

Reply:
Nice